### ISSUE TITLE
Replay on error from Get-UAL* cmdlets not functional

### ISSUE THREAD

**Issue by Qazeer** [invictus-ir/Microsoft-Extractor-Suite#16]
Hey! Reopening an issue on the absence of replay on error from `Search-UnifiedAuditLog` in `Get-UAL*` cmdlets, as the mentioned mechanism is not functional.

I can do a PR if you wish, but would need input on your prefered retry count.

---

I do understand the design choice, as it indeed complexify the code. Just a note however on the check for `Search-UnifiedAuditLog` not returning results, I think it is currently not working as intended (and thus why I had the error initially).

If we are talking about this one:

![image](https://github.com/invictus-ir/Microsoft-Extractor-Suite/assets/5614972/7df70867-082f-4c9c-9c0b-c6d81baf74e0)

`$retryCount` is not defined, and thus the `continue` statement is never reached.

Did a grep in a freshly cloned repo to make sure:
![image](https://github.com/invictus-ir/Microsoft-Extractor-Suite/assets/5614972/879215d0-283e-4a51-a0fb-b782fd40bf10)

_Originally posted by @Qazeer in https://github.com/invictus-ir/Microsoft-Extractor-Suite/issues/12#issuecomment-1608200887_

**Comment by JoeyInvictus** [comment]
Thank you for bringing this to our attention. It appears that an issue occurred during the transition from the old Microsoft 365 extractor to the Microsoft Extractor Suite. We have identified some improvements that can be made to the retry counter, but we will need to make "bigger" changes to the script in order to implement them.

We have scheduled an update within the next 1-3 weeks, and we will incorporate these changes into that release. We are eager to hear your feedback on the updated retry count functionality once the changes have been implemented.
